src/main/java/no/sikt/nva/pubchannels/handler/fetch/journal/FetchJournalByIdentifierAndYearHandler.java

```diff
@@ -38,9 +40,17 @@ protected FetchByIdAndYearResponse processInput(Void input, RequestInfo requestI
         var journalIdBaseUri = constructPublicationChannelIdBaseUri(JOURNAL_PATH_ELEMENT);
 
         var requestYear = request.getYear();
-        return FetchByIdAndYearResponse.create(journalIdBaseUri,
-                                               (ThirdPartyJournal) publicationChannelClient.getChannel(
-                                                   ChannelType.JOURNAL,
-                                                   request.getIdentifier(), requestYear), requestYear);
+        var journal = fetchJournal(request, requestYear);
+        return FetchByIdAndYearResponse.create(journalIdBaseUri, (ThirdPartyJournal) journal, requestYear);
+    }
+
+    private ThirdPartyPublicationChannel fetchJournal(FetchByIdAndYearRequest request, String requestYear)
+        throws ApiGatewayException {
+        try {
+            return publicationChannelClient.getChannel(ChannelType.JOURNAL, request.getIdentifier(), requestYear);
+        } catch (PublicationChannelMovedException movedException) {
+            throw new PublicationChannelMovedException(
+                "Journal moved", constructNewLocation(JOURNAL_PATH_ELEMENT, movedException.getLocation(), requestYear));
```

Comment:
Plan for handling this by consumers?

Reply:
Yes, https://unit.atlassian.net/browse/NP-46321